cibuildwheel/macos.py

```diff
@@ -579,17 +579,37 @@ def build(options: Options, tmp_path: Path) -> None:
                         shell_with_arch(before_test_prepared, env=virtualenv_env)
 
                     # install the wheel
+                    if is_cp38 and python_arch == "x86_64":
```

Comment:
It looks like this change only fixed the MacOS builds for the `cp38` case.  My builds for `cp36` and `cp37`, e.g.: https://github.com/alexlancaster/pypop/actions/runs/9866159045/job/27244457994#step:4:298 fail.

I had to modify my `pyproject.toml` to include the `SYSTEM_VERSION_COMPAT=0` as a workround, but I feel this should be in cibuildwheel, and the comment here: https://github.com/pypa/cibuildwheel/pull/1768#issuecomment-1970914303 seems to imply that the fix should also apply to <`cp38` builds too, but it doesn't appear to.

Reply:
Users of Python 3.6 and 3.7 can't install wheels that target macOS newer than 11.0. So you can build them, but users can't install them - well, they technically could, but they'd have to set SYSTEM_VERSION_COMPAT=0 too when they install them, otherwise pip will see the wheels as targeting a macOS that's newer than what's running.

Reply:
I see, but isn't it possible that Python 3.6 or 3.7 could be running on MacOS 12 or later? or did Python drop packages for those Python versions after 11.0?

Reply:
See this comment for further explanation- https://github.com/pypa/cibuildwheel/pull/1768#issuecomment-1970914303